### ISSUE TITLE
`RuntimeError` when closing a BNDB with BinSync connected

### ISSUE THREAD

**Issue by alexrp** [binsync/binsync#293]
### Description

```
Traceback (most recent call last):
  File "C:\Users\alex\AppData\Roaming\Binary Ninja\python310\site-packages\binsync\ui\utils.py", line 172, in run
    self._worker_thread()
  File "C:\Users\alex\AppData\Roaming\Binary Ninja\python310\site-packages\binsync\core\scheduler.py", line 60, in _worker_thread
    self._complete_a_job(block=True)
  File "C:\Users\alex\AppData\Roaming\Binary Ninja\python310\site-packages\binsync\core\scheduler.py", line 89, in _complete_a_job
    job.execute()
  File "C:\Users\alex\AppData\Roaming\Binary Ninja\python310\site-packages\binsync\core\scheduler.py", line 30, in execute
    self.ret_value = self.function(*self.args, **self.kwargs)
  File "C:\Users\alex\AppData\Roaming\Binary Ninja\python310\site-packages\binsync\api\controller.py", line 257, in _update_ui
    self.ui_callback(states)
  File "C:\Users\alex\AppData\Roaming\Binary Ninja\python310\site-packages\binsync\ui\control_panel.py", line 63, in update_callback
    self.update_ready.emit(status)
RuntimeError: Internal C++ object (ControlPanel) already deleted.
```

### Steps to reproduce the bug

1. Open a BNDB.
2. Connect to a BinSync repo.
3. Close the BNDB pane.

### Environment

Binary Ninja 3.5.4311-dev Personal (ff27871c)

**Comment by mahaloz** [comment]
@alexrp question for you and your use-case: when you close the BinSync control panel, do you expect the BinSync pull-push operations to stop altogether, or do you expect it to keep going in the background but just hide the UI?

**Comment by alexrp** [comment]
I'd expect it to keep going. My use case is doing mostly solo work on a binary and sharing the results to a public repo that others can consume if they wish. So I have little reason to keep the control panel open; it takes up too much valuable screen real estate (at least until it becomes a sidebar widget 🙂).

**Comment by mahaloz** [comment]
Closed by #287; however, this is still a problem for angr.
